openedx/features/course_card/helpers.py

```diff
@@ -101,18 +105,20 @@ def get_course_cards_list():
     return courses_list
 
 
-def is_course_in_programs(course_id):
+def is_course_in_programs(parent_course_key):
     """
-    Helper function to check if course is part of program
-    @param course_id: course key
-    @return: true if course is part of program otherwise false
+    Helper function to check if course is part of program.
+    @param parent_course_key: parent course key
+    @return: True if parent of any course reruns from discovery matches
+             with parent course key
     """
     programs = get_programs(get_current_request().site)
 
     for program in programs:
-        for program_course in program['courses']:
-            if program_course['course_runs']:
-                for course_rerun in program_course['course_runs']:
-                    if course_rerun['key'] == str(course_id):
-                        return True
+        for program_course in program.get('courses', []):
+            course_runs = program_course.get('course_runs')
+            if course_runs:
+                rerun_parent_course_key = get_related_card_id(course_runs[0].get('key'))
```

Comment:
It would be ideal case if we define a constant `FIRST_RERUN_INDEX` for `0` at the start of the body of this function.

Reply:
Yes I do agree with the second approach.